src/fairness/reader/test/data_reader_db_test01.cpp

```diff
@@ -36,14 +36,16 @@
 #include "src/common/libtap/tap.h"
 
 using namespace Flux::accounting;
+using namespace Flux::reader;
 
 static void test_fairshare_order (const std::string &filename,
                                   const std::vector<std::string> &expected)
 {
     bool bo = true;
     std::shared_ptr<weighted_tree_node_t> root;
+    data_reader_db_t data_reader;
```

Comment:
We only have one derived class so far, so this is probably fine, but once we have additional derived classes, we will probably need a factory class like in `resource`'s `reader` directory.

Reply:
Ah, thanks for pointing this out. What do you think about me posting a follow-up PR after this one that adds a factory class? I also don't mind adding another commit that adds one if we'd like to have it available from the start.

Reply:
> What do you think about me posting a follow-up PR after this one that adds a factory class? I can open an issue to track this.

Sounds good to me.